### ISSUE TITLE
Add translation strings to code

### ISSUE THREAD

**Issue by nadavkav** [h5p/h5p-php-report#2]
Hello H5P Team,
Please see if you can add new translation tokens (and strings) the the following hardcoded text labels in the report code, as we wish to translate them.

see links: https://github.com/h5p/h5p-php-report/blob/78233cf2810e6053d38de90e92b7a1b0f2b26e19/type-processors/matching-processor.class.php#L351-L353

Best,
Nadav

**Comment by timothyylim** [comment]
Hey @nadavkav :) 

I've created an issue for it here:

https://h5ptechnology.atlassian.net/browse/HFP-1875

PR's are accepted too 👍

**Comment by nadavkav** [comment]
Hi @timothyylim ,
Since h5p-php-report is a non standard Moodle "sub plugin", I can not just add a "lang" folder and have get_string() use it. I suggest updating h5p-moodle-plugin with a new db/subplugins.php file that looks like:
$subplugins = array('editor'=>'mod/hvp/editor', 'library'=>'mod/hvp/library', 'reporting' => 'mod/hvp/report');
And after this PR is accepted into h5p-moodle-plugin... I can add a lang folder to this "sub plug" and fix the missing tokens and initial English translation strings.
Do this plan make sense to you?

**Comment by timothyylim** [comment]
That makes sense to me, I've had the same problem with a new reporting template I've worked on recently. I kind of hacked it by passing the strings as html attributes: 

https://github.com/h5p/h5p-php-report/blob/HFP-1841-Add-reporting-OEQ/type-processors/iv-open-ended-question-processor.class.php#L36

@thomasmars, is one who understands this 'sub plugin' the most. Let's wait for his opinion first :)

**Comment by nadavkav** [comment]
Great :-)

**Comment by thomasmars** [comment]
To me it feels like we should implement a similar translation system to what H5P core is using, thus making use of the get_string() functionality in Moodle and define the strings at the 'h5p' component level. E.g. reporting would have an interface for a translation function, then the H5P moodle plugin would implement the reporting module interface and thus also the translation function by using get_string(), or the H5P moodle plugin could pass in a dictionary to the reporting module containing all the strings that the reporting module needed.

The reporting module is a submodule that is used in other contexts than Moodle, so the translation system must be decoupled from Moodle and the subplugin system that Moodle has. However it could be wrapped in a Moodle submodule, but this would result in the H5P Moodle plugin having a Moodle reporting submodule, which then again would wrap the reporting module. The addition of the Moodle submodule layer seems unnecessary, but if there are compelling arguments for structuring it like this we should look closer at it.

**Comment by nadavkav** [comment]
> Implement a similar translation system to what H5P core is using, thus making use of the get_string() functionality in Moodle and define the strings at the 'h5p' component level.

I agree, since this reporting module is shared with other non Moodle systems, it should be exposing an interface for translation, as other sub modules should.
Can you put it on the road map? open an issue for it on the jira?

**Comment by thomasmars** [comment]
I've updated the [issue ](https://h5ptechnology.atlassian.net/browse/HFP-1875)Timothy created in the jira issue tracker. Thanks for the input.

**Comment by otacke** [comment]
@nadavkav Long time no see! I was just made a community maintainer, and I'll try to help giving all the open issues and pull requests some closure. Eventually.

I noticed that [HFP-1875](https://h5ptechnology.atlassian.net/browse/HFP-1875) was never worked on, unfortunately. Also, I cannot tell if and when it will - but I know that the topic of "reportings" is on H5P Group's agenda (cmp. https://vimeo.com/1198738965?fl=pl&fe=sh#t=20m19s). I'll try to give that ticket some attention.

Absolutely not the way this should have played out in the past, but wanted to let you know that "Nothing is forgotten. Nothing is ever forgotten." (quoting Robin Hood from the 80s series "Robin of Sherwood" that I just watched ;-))

[HFP-1875]: https://h5ptechnology.atlassian.net/browse/HFP-1875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
